docs/topics/advent-of-code.md

```diff
@@ -0,0 +1,80 @@
+[//]: # (title: Advent of Code puzzles in idiomatic Kotlin)
+
+[Advent of Code](https://adventofcode.com/) is an annual series of Christmas-themed puzzles that follow an Advent calendar,
```

Comment:
While it's pretty obvious from the name, let's substitute "Christmas" with "holiday".

Reply:
We can also just say that it's an event happening in December where new puzzles are published from 1 Dec to 25 Dec, for example.